x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/PutRoleRequestBuilderFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.role;
+
+import org.elasticsearch.client.internal.Client;
+
+import java.util.function.Predicate;
+
+public interface PutRoleRequestBuilderFactory {
+    PutRoleRequestBuilder create(Client client, boolean restrictRequest, Predicate<String> fileRolesStoreNameChecker);
+
+    class Default implements PutRoleRequestBuilderFactory {
+        @Override
+        public PutRoleRequestBuilder create(Client client, boolean restrictRequest, Predicate<String> fileRolesStoreNameChecker) {
+            return new PutRoleRequestBuilder(client);
```

Comment:
nit: would be helpful to add a brief javadoc on why the default implementation ignores restrictions